include/swift/Basic/Semantics.def

```diff
@@ -0,0 +1,78 @@
+//===--- Semantics.def - Semantics Attribute Definitions -------------*- C++ -*-===//
```

Comment:
This (and the corresponding header) should probably go in `AST/` as opposed to `Basic` so it's in the same lib as `hasSemanticsAttr`

`Semantics.def` Also feels a bit vague to me. What do you think about `SemanticAttrs.def`?